helm/external-dns-app/templates/deployment.yaml

```diff
@@ -33,15 +33,29 @@ spec:
         runAsGroup: {{ .Values.global.securityContext.groupID }}
         fsGroup: {{ .Values.global.securityContext.fsGroupID }}
       priorityClassName: giantswarm-critical
-      {{- if and (eq .Values.provider "aws") (eq .Values.aws.access "internal") }}
       initContainers:
+      {{- if and (eq .Values.provider "aws") (eq .Values.aws.access "internal") }}
       - name: wait-for-iam-role
         image: {{ .Values.global.image.registry }}/giantswarm/alpine:3.12.0
         command:
         - /bin/sh
         - -c
         - counter=5; while ! wget -qO- http://169.254.169.254/latest/meta-data/iam/security-credentials/ | grep {{ template "aws.iam.role" . }}; do echo 'Waiting for iam-role to be available...'; sleep 5; let "counter-=1"  ; if [ "$counter" -eq "0" ]; then exit 1; fi; done
       {{- end }}
+      {{- if eq .Values.provider "azure" }}
+      - name: copy-azure-config-file
+        image: {{ .Values.global.image.registry }}/giantswarm/alpine:3.12-python3
+        command:
+          - /bin/sh
+          - -c
+          - if [ -f /etc/kubernetes/config/azure.yaml ]; then cp /etc/kubernetes/config/azure.yaml /config/azure.yaml; else cat /etc/kubernetes/azure.json | python3 -c 'import sys, yaml, json; print(yaml.dump(json.loads(sys.stdin.read())))' > /config/azure.yaml; fi
```

Comment:
Can you make this more readable and add a comment what this does and where the files come from?